substrate-node/pallets/pallet-smart-contract/src/storage_state/v9.rs

```diff
@@ -0,0 +1,978 @@
+                    );
+                }
+            }
+        } else {
+            debug!(
+                " ⚠️    ContractIDByNodeIDAndHash[node: {}, hash: {:?}]: contract {} not exists",
+                node_id, String::from_utf8_lossy(&hash), contract_id
+            );
+        }
+    }
+
+    debug!(
+        "🏁  Smart Contract pallet {:?} checking ContractIDByNodeIDAndHash storage map END",
+        PalletVersion::<T>::get()
+    );
+}
+
+// ContractIDByNameRegistration
+pub fn check_contract_id_by_name_registration<T: Config>() {
+    debug!(
+        "🔎  Smart Contract pallet {:?} checking ContractIDByNameRegistration storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    for (name, contract_id) in ContractIDByNameRegistration::<T>::iter() {
+        if let Some(c) = Contracts::<T>::get(contract_id) {
+            match c.contract_type {
+                types::ContractData::NameContract(name_contract) => {
+                    if name_contract.name != name {
+                        debug!(
+                            " ⚠️    ContractIDByNameRegistration[name: {:?}]: name ({:?}) on contract {} is not matching",
+                            String::from_utf8_lossy(&name.into()), String::from_utf8_lossy(&name_contract.name.into()), contract_id, 
+                        );
+                    }
+                }
+                _ => {
+                    debug!(
+                        " ⚠️    ContractIDByNameRegistration[name: {:?}]: contract {} is not a name contract",
+                        String::from_utf8_lossy(&name.into()), contract_id
+                    );
+                }
+            }
+        } else {
+            debug!(
+                " ⚠️    ContractIDByNameRegistration[name: {:?}]: contract {} not exists",
+                String::from_utf8_lossy(&name.into()), contract_id
+            );
+        }
+    }
+
+    debug!(
+        "🏁  Smart Contract pallet {:?} checking ContractIDByNameRegistration storage map END",
+        PalletVersion::<T>::get()
+    );
+}
+
+// ContractLock
+pub fn check_contract_lock<T: Config>() {
+    debug!(
+        "🔎  Smart Contract pallet {:?} checking ContractLock storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    for (contract_id, _contract_lock) in ContractLock::<T>::iter() {
+        if Contracts::<T>::get(contract_id).is_none() {        
+            debug!(
+                " ⚠️    ContractLock[contract: {}]: contract not exists",
+                contract_id
+            );
+        }
+    }
+
+    debug!(
+        "🏁  Smart Contract pallet {:?} checking ContractLock storage map END",
+        PalletVersion::<T>::get()
+    );
+}
+
+// SolutionProviders
+pub fn check_solution_providers<T: Config>() {
+    debug!(
+        "🔎  Smart Contract pallet {:?} checking SolutionProviders storage map START",
+        PalletVersion::<T>::get()
+    );
+
+    let solution_provider_id_range = 1..=SolutionProviderID::<T>::get();
```

Comment:
Would this not work:
```
let current_solution_provider_id = SolutionProviderID::<T>::get();
for (solution_provider_id, _solution_provider) in SolutionProviders::<T>::iter() {
        if solution_provider_id >= current_solution_provider_id {
         ...

```

Reply:
Same as https://github.com/threefoldtech/tfchain/pull/659/files/cc15fc61e7ef318f015e7434c1a88928a12833dc#r1201362350
also want to avoid case when  `solution_provider_id == 0`